pkg/kv/kvclient/rangefeed/rangefeed.go

```diff
@@ -196,23 +196,27 @@ func (f *RangeFeed) Start(ctx context.Context, spans []roachpb.Span) error {
 		return errors.AssertionFailedf("expected at least 1 span, got none")
 	}
 
-	if !atomic.CompareAndSwapInt32(&f.started, 0, 1) {
-		return errors.AssertionFailedf("rangefeed already started")
-	}
-
 	// Maintain a frontier in order to resume at a reasonable timestamp.
 	// TODO(ajwerner): Consider exposing the frontier through a RangeFeed method.
 	// Doing so would require some synchronization.
 	frontier, err := span.MakeFrontier(spans...)
 	if err != nil {
 		return err
 	}
+	return f.start(ctx, frontier, true)
+}
 
-	for _, sp := range spans {
-		if _, err := frontier.Forward(sp, f.initialTimestamp); err != nil {
-			frontier.Release() // release whatever was allocated.
-			return err
-		}
+// StartFromFrontier is like Start but allows passing a frontier to reflect any
+// previous progress. The rangefeed takes ownership of the passed frontier until
+// it is closed; the caller must not interact with it until Close returns. The
+// caller remains responsible for releasing the frontier thereafter however.
```

Comment:
This is hopefully obvious from the signature, but it is also required that the frontier exactly covers the spans you want to follow with the rangefeed.

Reply:
Updated the comment to say so